packages/nyc-test-coverage/src/interceptors.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  proxySinks,
+  SignalInput,
+  WorkflowExecuteInput,
+  WorkflowInterceptorsFactory,
+  WorkflowInboundCallsInterceptor,
+  QueryInput
+} from '@temporalio/workflow';
+import { CoverageSinks } from './sinks';
+
+const { coverage } = proxySinks<CoverageSinks>();
+
+class CoverageInterceptor implements WorkflowInboundCallsInterceptor {
+  public async handleQuery(input: QueryInput, next: any): Promise<unknown> {
+    const ret = await next(input);
+
+    // @ts-ignore
+    const globalCoverage = global.__coverage__;
+    coverage.merge(JSON.parse(JSON.stringify(globalCoverage)));
+    clearCoverage(globalCoverage);
```

Comment:
Why clear?

Reply:
Same as https://github.com/temporalio/sdk-typescript/pull/798#discussion_r934087391 : if we don't clear coverage after merging, we start double-counting line and function executions.